imports/plugins/core/orders/server/no-meteor/mutations/cancelOrderItem.js

```diff
@@ -92,11 +92,15 @@ export default async function cancelOrderItem(context, input) {
       // It will have the same status as this item has before we cancel it.
       // Later, after we exit this loop, we'll append it to the `group.items` list.
       if (item.quantity > cancelQuantity) {
+        const newItemQuantity = item.quantity - cancelQuantity;
         itemToAdd = {
           ...item,
           _id: Random.id(),
-          quantity: item.quantity - cancelQuantity
+          quantity: newItemQuantity
         };
+
+        // Update the subtotal since it is related to the quantity
+        itemToAdd.subtotal = item.price.amount * newItemQuantity;
```

Comment:
Haven't tested this in action yet, but based on fixing a similar issue when adding items to cart, I think that `subtotal` is an object that contains an `amount` and `currencyCode`, not just a `number`.

Will confirm when testing...